File: 1528-kids-with-the-greatest-number-of-candies/kids-with-the-greatest-number-of-candies.py
class Solution(object):
    def kidsWithCandies(self, candies, extraCandies):
        a=[]
        b=[]
        for i in candies:
            a.append(i)
            maxm= max(a)
        for j in range(len(candies)):
            if candies[j]+extraCandies>=maxm:
                result= True
                b.append(result)
            else:
                result= False 
                b.append(result)
        return b    



        """
        :type candies: List[int]
        :type extraCandies: int
        :rtype: List[bool]
        """
        

File: 1528-kids-with-the-greatest-number-of-candies/kids-with-the-greatest-number-of-candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        int maxm = candies[0];
        List<Boolean> result = new ArrayList<>();
        
        for (int i = 1; i < candies.length; i++) {
            maxm = Math.max(maxm, candies[i]);
        }
        
        for (int j = 0; j < candies.length; j++) {
            if (candies[j] + extraCandies >= maxm) {
                result.add(true);
            } else {
                result.add(false);
            }
        }
        
        return result;
    }
}